src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs

```diff
@@ -40,67 +42,212 @@ public FeedOptions FeedOptions
+            var tempFolderFullPath = Path.Combine(installFolder, tempFolderName);
+
+            // Delete if exists already
+            FileOperationUtils.DeleteFolder(tempFolderFullPath);
+            Directory.CreateDirectory(tempFolderFullPath);
 
-            RestoreCommand.RestorePackageId = packageId;
-            RestoreCommand.RestorePackageVersion = packageVersion;
+            WriteVerbose("Temporary folder name: {0}", tempFolderFullPath);
+            var projectFileFullPath = Path.Combine(tempFolderFullPath, "project.json");
 
-            return
-                await RestoreCommand.ExecuteCommand() &&
-                InstallCommands(packageId, RestoreCommand.AppInstallPath);
+            File.WriteAllText(
+                projectFileFullPath,
+                string.Format(
+        @"{{
+    ""dependencies"":{{
+        ""{0}"":""{1}""
+    }}
+}}", packageName, packageVersion));
+
+            return projectFileFullPath;
         }
 
-        private bool InstallCommands(string packageId, string appPath)
+        private bool ValidateApplicationPackage(string appFolderFullPath)
         {
-            string commandsFolder = Path.Combine(appPath, InstallBuilder.CommandsFolderName);
+            string commandsFolder = Path.Combine(appFolderFullPath, InstallBuilder.CommandsFolderName);
 
             if (!Directory.Exists(commandsFolder))
             {
-                LogError("The specified package is not an application. The package was added but no commands were installed.");
+                WriteError("The specified package is not an application. The package was added but no commands were installed.");
                 return false;
             }
 
-            var fileFilter = PlatformHelper.IsWindows ? "*.cmd" : "*.sh";
-            var allAppCommandsFiles = Directory.EnumerateFiles(commandsFolder, fileFilter);
-            var allAppCommands = allAppCommandsFiles
-                .Select(cmd => Path.GetFileNameWithoutExtension(cmd))
-                .ToList();
-
-            var blockedCommands = _commandsRepository.Commands.Where(cmd => 
+            var blockedCommands = _commandsRepository.Commands.Where(cmd =>
                 !CommandNameValidator.IsCommandNameValid(cmd) ||
                 CommandNameValidator.ShouldNameBeSkipped(cmd));
             if (blockedCommands.Any())
             {
-                LogError(string.Format(
+                WriteError(string.Format(
                     "The application cannot be installed because the following command names are not allowed: {0}.",
                     string.Join(", ", blockedCommands)));
 
                 return false;
             }
 
+            return true;
+        }
+
+        private bool InstallCommands(string packageId, string appPath)
+        {
+            string commandsFolder = Path.Combine(appPath, InstallBuilder.CommandsFolderName);
+
+            var fileFilter = PlatformHelper.IsWindows ? "*.cmd" : "*.sh";
```

Comment:
Ah, I remember why I added the `.sh` extension. The NuGet package builder doesn't support file without extensions. I'm not sure if it is a bug or not 

```
System.ArgumentOutOfRangeException: startIndex cannot be larger than length of string.
Parameter name: startIndex
   at System.String.Substring(Int32 startIndex, Int32 length)
   at System.String.Substring(Int32 startIndex)
   at NuGet.PackageBuilder.WriteFiles(ZipArchive package) in D:\aspnet\dnx\src\Microsoft.Framework.Runtime\NuGet\Authori
ng\PackageBuilder.cs:line 468
   at NuGet.PackageBuilder.Save(Stream stream) in D:\aspnet\dnx\src\Microsoft.Framework.Runtime\NuGet\Authoring\PackageB
uilder.cs:line 257
   at Microsoft.Framework.PackageManager.BuildManager.Build() in D:\aspnet\dnx\src\Microsoft.Framework.PackageManager\Bu
ilding\BuildManager.cs:line 184
   at Microsoft.Framework.PackageManager.Program.<>c__DisplayClass3_3.<Main>b__6() in D:\aspnet\dnx\src\Microsoft.Framew
ork.PackageManager\Program.cs:line 162
   at Microsoft.Framework.Runtime.Common.CommandLine.CommandLineApplication.Execute(String[] args) in D:\aspnet\dnx\src\
Microsoft.Framework.CommandLineUtils\CommandLine\CommandLineApplication.cs:line 277
   at Microsoft.Framework.PackageManager.Program.Main(String[] args) in D:\aspnet\dnx\src\Microsoft.Framework.PackageMan
```